hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/SnapshotScannerHDFSAclController.java

```diff
@@ -263,14 +262,29 @@ public void postDeleteTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
     }
   }
 
+  @Override
+  public void postModifyTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
+      TableName tableName, TableDescriptor oldDescriptor, TableDescriptor currentDescriptor)
+      throws IOException {
+    if (needHandleTableHdfsAcl(currentDescriptor, "modifyTable " + tableName)
+        && !hdfsAclHelper.isTableUserScanSnapshotEnabled(oldDescriptor)) {
+      hdfsAclHelper.createTableDirectories(tableName);
+      Set<String> tableUsers = hdfsAclHelper.getUsersWithTableReadAction(tableName, false, false);
+      Set<String> users =
+          hdfsAclHelper.getUsersWithNamespaceReadAction(tableName.getNamespaceAsString(), true);
+      users.addAll(tableUsers);
+      hdfsAclHelper.addTableAcl(tableName, users);
+      SnapshotScannerHDFSAclStorage.addUserTableHdfsAcl(ctx.getEnvironment().getConnection(),
+        tableUsers, tableName);
+    }
```

Comment:
How it will happen if we have an TableUserScanSnapshot Enabled table, and then modify to disabled ?  I think we shold also remove the HDFS ACL ?